### ISSUE TITLE
Unable to launch tensorboard (Error in tensorboard)

### ISSUE THREAD

**Issue by urhero** [rstudio/tensorflow#227]
tensorflow-1.5.0
pip-9.0.2
R-3.4.3(64bit)
python-3.6.4
Using Rstudio in Anaconda r-tensorflow environment.

Hi, 
I am having the following error the whole day....

'>tensorboard("C:/Users/IKM/AppData/Local/Temp/2/tmpcc3t2288", launch_browser = TRUE)
Error in tensorboard("C:/Users/IKM/AppData/Local/Temp/2/tmpcc3t2288",  : 
  Unable to launch tensorboard

The full code is from
https://www.edgarsdatalab.com/2017/10/22/intro-to-tensorflow-in-r/#fullcode

**Comment by jjallaire** [comment]
There has recently been some breakage when updating tensorboard within an existing environment. I'd recommend starting by removing the r-tensorflow environment entirely then re-installing, e.g.

```r
library(reticulate)
conda_remove("r-tensorflow")
library(tensorflow)
install_tensorflow()
```

**Comment by faltinl** [comment]
Same problem here. I did the re-installing sequence as indicated, everything worked smoothly:

`> install_tensorflow(method = "conda", version = "1.5.1")`
Using r-tensorflow conda environment for TensorFlow installation
Solving environment: ...working... done

**Package Plan** 
  environment location: C:\Users\lf\ANACON~1\envs\r-tensorflow
  **added / updated specs:** 
    - h5py
    - pillow
    - pyyaml
    - requests
    - scipy
    - tensorflow==1.5.1

**The following packages will be downloaded:**
    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2018.4.16  |                0         176 KB  conda-forge
    numpy-1.13.3               |   py36h5c71026_4         3.7 MB
    asn1crypto-0.24.0          |           py36_0         155 KB  conda-forge
    pyopenssl-17.5.0           |           py36_1          82 KB  conda-forge
    idna-2.6                   |           py36_1         122 KB  conda-forge
    chardet-3.0.4              |           py36_0         210 KB  conda-forge
    intel-openmp-2018.0.0      |                8         1.4 MB
    certifi-2018.4.16          |           py36_0         143 KB  conda-forge
    cffi-1.11.5                |           py36_0         214 KB  conda-forge
    cryptography-2.2.1         |           py36_0         350 KB  conda-forge
    urllib3-1.22               |           py36_0         154 KB  conda-forge
    pycparser-2.18             |           py36_0         169 KB  conda-forge
    webencodings-0.5.1         |           py36_0          20 KB  conda-forge
    win_inet_pton-1.0.1        |           py36_1           5 KB  conda-forge
    olefile-0.45.1             |           py36_0          50 KB  conda-forge
    pysocks-1.6.8              |           py36_1          23 KB  conda-forge
    six-1.11.0                 |           py36_1          21 KB  conda-forge
    mkl_fft-1.0.2              |           py36_0         114 KB  conda-forge
    mkl_random-1.0.1           |           py36_0         267 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         7.3 MB

**The following NEW packages will be INSTALLED:**
    absl-py:           0.1.10-py_0           conda-forge
    asn1crypto:        0.24.0-py36_0         conda-forge
    backports:         1.0-py36_1            conda-forge
    backports.weakref: 1.0rc1-py36_1         conda-forge
    bleach:            1.5.0-py36_0          conda-forge
    ca-certificates:   2018.4.16-0           conda-forge
    cffi:              1.11.5-py36_0         conda-forge
    chardet:           3.0.4-py36_0          conda-forge
    cryptography:      2.2.1-py36_0          conda-forge
    freetype:          2.8.1-vc14_0          conda-forge [vc14]
    h5py:              2.7.1-py36_3          conda-forge
    hdf5:              1.10.1-vc14_2         conda-forge [vc14]
    html5lib:          0.9999999-py36_0      conda-forge
    icc_rt:            2017.0.4-h97af966_0              
    idna:              2.6-py36_1            conda-forge
    intel-openmp:      2018.0.0-8                       
    jpeg:              9b-vc14_2             conda-forge [vc14]
    libpng:            1.6.34-vc14_0         conda-forge [vc14]
    libprotobuf:       3.5.2-vc14_0          conda-forge [vc14]
    libtiff:           4.0.9-vc14_0          conda-forge [vc14]
    markdown:          2.6.11-py_0           conda-forge
    mkl:               2018.0.2-1                       
    mkl_fft:           1.0.2-py36_0          conda-forge
    mkl_random:        1.0.1-py36_0          conda-forge
    numpy:             1.13.3-py36h5c71026_4            
    olefile:           0.45.1-py36_0         conda-forge
    openssl:           1.0.2o-vc14_0         conda-forge [vc14]
    pillow:            5.1.0-py36_0          conda-forge
    protobuf:          3.5.2-py36_vc14_0     conda-forge [vc14]
    pycparser:         2.18-py36_0           conda-forge
    pyopenssl:         17.5.0-py36_1         conda-forge
    pyreadline:        2.1-py36h094d952_1               
    pysocks:           1.6.8-py36_1          conda-forge
    pyyaml:            3.12-py36_1           conda-forge
    requests:          2.18.4-py36_1         conda-forge
    scipy:             1.0.1-py36hce232c7_0             
    six:               1.11.0-py36_1         conda-forge
    **tensorboard:       1.5.1-py36_1          conda-forge
    tensorflow:        1.5.1-py36_0          conda-forge**
    tk:                8.6.7-vc14_0          conda-forge [vc14]
    urllib3:           1.22-py36_0           conda-forge
    webencodings:      0.5.1-py36_0          conda-forge
    werkzeug:          0.14.1-py_0           conda-forge
    win_inet_pton:     1.0.1-py36_1          conda-forge
    yaml:              0.1.7-vc14_0          conda-forge [vc14]
    zlib:              1.2.11-vc14_0         conda-forge [vc14]

**The following packages will be UPDATED:**
    certifi:           2018.4.16-py36_0                  --> 2018.4.16-py36_0 conda-forge

**Downloading and Extracting Packages**
ca-certificates 2018.4.16########## | 100% 
numpy 1.13.3########## | 100% 
asn1crypto 0.24.0########## | 100% 
pyopenssl 17.5.0########## | 100% 
idna 2.6########## | 100% 
chardet 3.0.4########## | 100% 
intel-openmp 2018.0.0########## | 100% 
certifi 2018.4.16########## | 100% 
cffi 1.11.5########## | 100% 
cryptography 2.2.1########## | 100% 
urllib3 1.22########## | 100% 
pycparser 2.18########## | 100% 
webencodings 0.5.1########## | 100% 
win_inet_pton 1.0.1########## | 100% 
olefile 0.45.1########## | 100% 
pysocks 1.6.8########## | 100% 
six 1.11.0########## | 100% 
mkl_fft 1.0.2########## | 100% 
mkl_random 1.0.1########## | 100% 
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
**Installation complete.**
Restarting R session...

However...
`> tensorboard("C:/Users/lf/Documents/R/logging", reload_interval = 1)`  
(the file name given exists and has been used successfully with/by my previous tensorflow installation) led to... 
```
Error in tensorboard("C:/Users/lf/Documents/R/logging", reload_interval = 1) : 
  Unable to launch tensorboard
```
The only hint that something might be fishy: my package list in R_Studio still shows "tensorflow ... 1.5.0.9000" and not "1.5.1". Any idea what could be wrong?

**Comment by bmenze** [comment]
Same problem here. I followed the reinstall advice, but problem persists:
`Traceback (most recent call last):
File "C:\ProgramData\Miniconda3\envs\r-tensorflow\Scripts\tensorboard-script.py", line 3, in <module>
import tensorflow.tensorboard.tensorboard
ModuleNotFoundError: No module named 'tensorflow.tensorboard'`

**Comment by faltinl** [comment]
Yes, what a pity. Knowing that there is a useful instrument but not being able to actally use it is definitely frustrating. Compatibility of several different components seems to be crucial, since during my many attempts to reinstall everything I got a host of different and sometimes contradictory error messages. 

I must confess, I have given up. Perhaps with future versions of R, RStudio, Keras, Python, Tensorflow, Anaconda etc. the situation might improve - whatever....

**Comment by bmenze** [comment]
I've now installed R/Rstudio/Python/Keras/Tensorflow on several other machines, both Windows and Ubuntu, with and without previous installations of Keras and Anaconda and Python, but to no avail. 

Different versions of the tensorboard error about the visibility of the required python / tensorflow packages remain:  "ModuleNotFoundError: No module named 'tensorflow.tensorboard'". (If someone is interested in reproducibility: In all those cases I was testing the Unet example code for Windows/Unix.)

I agree, it's a pity, and (re-)turning to Python sounds like the best option. Maybe dedicated installation instructions would help - I noticed there were some related threads about installation / visibility issues before...

**Comment by fengqifang** [comment]
@faltinl 

I got the same problem today. 

First, I have checked my anaconda environment: r-tensorflow, I found that I do have tensorboard module, but no tensorflow.tensorboard. 

Then, based on the error message, I have edited the 'tensorboard-script.py' script located in '../envs/r-tensorflow/Scripts/ as blow

```{python}
if __name__ == '__main__':
    import sys
    import tensorboard.main

    sys.exit(tensorboard.main.main())
```

Then I can launch the TensorBoard. Hope that helps.
